src/main/com/intellij/lang/jsgraphql/psi/GraphQLPsiUtil.java

```diff
@@ -48,4 +53,54 @@ public static String getTypeName(PsiElement psiElement, @Nullable Ref<GraphQLIde
         return null;
     }
 
+    @Nullable
+    public static VirtualFile getVirtualFileFromPsiFile(@Nullable PsiFile containingFile) {
```

Comment:
com.intellij.psi.util.PsiUtilCore#getVirtualFile ?

Reply:
It does
```
public static VirtualFile getVirtualFile(@Nullable PsiElement element) {
    // optimisation: call isValid() on file only to reduce walks up and down
    if (element == null) {
      return null;
    }
    if (element instanceof PsiFileSystemItem) {
      return element.isValid() ? ((PsiFileSystemItem)element).getVirtualFile() : null;
    }
```
and it differs from the existing behavior here.